### DISCUSSION TITLE
Auto-scroll as we move an element offscreen

### DISCUSSION THREAD

**Discussion by seekeramento** [clientIO/joint#1773] [Q&A] [answered]
When moving an element (node) offscreen, the dragging will stop the point we become not able to move any further off the canvas.
Is there a ready API in JointJS that automatically scrolls towards the direction of dragging while expanding the viewport as required?

This will require some efforts but wand to see if there are already some APIs or examples to look into first.

**Comment by jamesgeorgewilliams** [comment, marked as answer]
If trying to implement it yourself in JointJS, you could take advantage of  `paper.fitToContent()` .
https://resources.jointjs.com/docs/jointjs/v3.5/joint.html#dia.Paper.prototype.fitToContent

If using JointJS+, the PaperScroller plugin has an `autoResizePaper` option.
https://resources.jointjs.com/docs/rappid/v3.5/ui.PaperScroller.html

There is also a `scrollWhileDragging` option that can be used along with `autoResizePaper` in the PaperScroller.

**Reply by seekeramento** [reply]
Yes, I am using JointJS+.
I was able to use paper.fitToContent but this does not auto-scrolll as the element is being moved off-screen. 
It seems that scrollWhileDragging is what I am looking for. Is there any demo or example that uses it?

**Reply by jamesgeorgewilliams** [reply]
Chatbot, BPMN Editor, KitchenSink and Planogram all use `scrollWhileDragging: true`
https://resources.jointjs.com/demos/chatbot
https://resources.jointjs.com/demos/bpmn
https://resources.jointjs.com/demos/kitchensink
https://resources.jointjs.com/demos/planogram

**Reply by seekeramento** [reply]
Thanks a lot. This is helpful.
